torch/fx/graph.py

```diff
@@ -252,6 +252,288 @@ class _PyTreeInfo(NamedTuple):
     in_spec: pytree.TreeSpec
     out_spec: Optional[pytree.TreeSpec]
 
+@compatibility(is_backward_compatible=False)
+class CodeGen(object):
+    def __init__(self):
+        self._body_transformer: Optional[TransformCodeFunc] = None
+
+    def generate_prologue(self, free_vars: List[str], maybe_return_annotation: str) -> str:
+        """
+        Given the free variables and a return annotation, generates the beginning of the FX function.
+        By default, `generate_prologue(['a', 'b'], '') == 'def forward(a, b):'`
+        """
+        # If the original function didn't have self as its first argument, we
+        # would have added it.
+        if len(free_vars) == 0 or free_vars[0] != 'self':
+            free_vars.insert(0, 'self')
```

Comment:
I wonder if we can hoist this into the calling context. Seems like a strange bit of boilerplate each override of this function would have to copy/paste

Reply:
I think this is tricky, since the current API involves being given the `free_vars`, and then generating the function signature directly. And most of the custom codegen probably doesn't want to handle the `self` argument along with the actual inputs to the function.

I think the intended idea is to call `super().gen_code_fn` with the updated `free_vars`.